src/main/java/net/neoforged/neoforge/registries/RegistryBuilder.java

```diff
@@ -5,306 +5,103 @@
-        if (clearCallback.size() == 1)
-            return clearCallback.get(0);
-
-        return (owner, stage) -> {
-            for (IForgeRegistry.ClearCallback<T> cb : this.clearCallback)
-                cb.onClear(owner, stage);
-        };
-    }
-
-    @Nullable
-    public IForgeRegistry.CreateCallback<T> getCreate() {
-        if (createCallback.isEmpty())
-            return null;
-        if (createCallback.size() == 1)
-            return createCallback.get(0);
-
-        return (owner, stage) -> {
-            for (IForgeRegistry.CreateCallback<T> cb : this.createCallback)
-                cb.onCreate(owner, stage);
-        };
-    }
-
-    @Nullable
-    public IForgeRegistry.ValidateCallback<T> getValidate() {
-        if (validateCallback.isEmpty())
-            return null;
-        if (validateCallback.size() == 1)
-            return validateCallback.get(0);
-
-        return (owner, stage, id, key, obj) -> {
-            for (IForgeRegistry.ValidateCallback<T> cb : this.validateCallback)
-                cb.onValidate(owner, stage, id, key, obj);
-        };
-    }
-
-    @Nullable
-    public IForgeRegistry.BakeCallback<T> getBake() {
-        if (bakeCallback.isEmpty())
-            return null;
-        if (bakeCallback.size() == 1)
-            return bakeCallback.get(0);
-
-        return (owner, stage) -> {
-            for (IForgeRegistry.BakeCallback<T> cb : this.bakeCallback)
-                cb.onBake(owner, stage);
-        };
-    }
-
-    @Nullable
-    public ResourceLocation getDefault() {
-        return this.optionalDefaultKey;
-    }
-
-    public int getMinId() {
-        return minId;
-    }
-
-    public int getMaxId() {
-        return maxId;
-    }
-
-    public boolean getAllowOverrides() {
-        return allowOverrides;
-    }
-
-    public boolean getAllowModifications() {
-        return allowModifications;
-    }
-
-    @Nullable
-    public IForgeRegistry.MissingFactory<T> getMissingFactory() {
-        return missingFactory;
-    }
-
-    public boolean getSaveToDisc() {
-        return saveToDisc;
-    }
-
-    public boolean getSync() {
-        return sync;
-    }
-
-    public Set<ResourceLocation> getLegacyNames() {
-        return legacyNames;
-    }
-
-    Function<T, Holder.Reference<T>> getIntrusiveHolderCallback() {
-        return this.intrusiveHolderCallback;
-    }
-
-    boolean getHasWrapper() {
-        return this.hasWrapper;
+    public Registry<T> create() {
+        BaseMappedRegistry<T> registry = this.defaultKey != null
+                ? new DefaultedMappedRegistry<>(this.defaultKey.toString(), this.registryKey, Lifecycle.stable(), false)
+                : new MappedRegistry<>(this.registryKey, Lifecycle.stable(), false);
+        this.callbacks.forEach(registry::addCallback);
+        if (this.maxId != -1)
+            registry.setMaxId(this.maxId);
+        registry.setSync(this.sync);
+
+        RegistryManager.trackModdedRegistry(registry.key().location());
```

Comment:
Some people (@lukebemish) are not happy about this - see discussion on Discord: https://discord.com/channels/313125603924639766/1131130393144336385/1173667717899898921. Personally I think the safeguard is useful, but I'll let others decide.

Reply:
Added a builder option (`disableRegistrationCheck`) to opt out of these checks, for advanced/niche use cases that don't want them.